### PR TITLE
add more examples[support dynamic, static and aware_training quantization]

### PR DESCRIPTION
**Pull request by xin3he**
# What does this PR do?

This PR will provide more available tasks for user to apply. Tasks can be apply with dynamic, static or aware_training quantization with INC.

## Task includes:
- [multiple-choice] Done
- [summarization] Done
- [language-modeling]  CLM Done, MLM&PLM Done.
  - limitation: a seed set in MLM&PLM `take_eval_step` to ensure verify loading success.
- [translation] Done.
[issues#14268](https://github.com/huggingface/transformers/issues/14268),  [PR#14276](https://github.com/huggingface/transformers/pull/14276) fixed the issue.

### CODE REVIEW

File: src/optimum/intel/neural_compressor/utils.py [hunk 1]
```diff
@@ -34,8 +34,6 @@ def from_pytorch_dataloader(cls, dataloader: DataLoader):
 
     def __iter__(self):
         for input in super().__iter__():
-            if not isinstance(input, (dict, tuple, list)):
-                raise TypeError(f"Model calibration cannot use input of type {type(input)}.")
```
Comment (xin3he) [inline src/optimum/intel/neural_compressor/utils.py:38]: I delete them because model calibration cannot use input of type <class 'transformers.tokenization_utils_base.BatchEncoding'> in summarization task.

**Comment by xin3he** [comment]
Is there a test for examples to trigger?
What should I do to make it happen?
BTW, [language-modeling] and [translation] haven't ready now. Please wait, thanks.

File: examples/pytorch/language-modeling/run_clm.py [hunk 2]
```diff
@@ -0,0 +1,637 @@
...
+        training_args.eval_steps = 100
+        trainer.args.metric_for_best_model = metric_name.split("_")[1]
+        trainer.args.greater_is_better = True
+        trainer.args.load_best_model_at_end = True
+        trainer.args.evaluation_strategy = IntervalStrategy.STEPS
+
+        early_stopping_patience = 2
+        early_stopping_threshold = 0.001 # optional
+        trainer.add_callback(transformers.EarlyStoppingCallback(early_stopping_patience, \
+                                                                early_stopping_threshold))
```
[the remark this answers is not in the document]
Comment (xin3he) [reply examples/pytorch/language-modeling/run_clm.py:534]: Sure, will remove.

File: examples/pytorch/language-modeling/run_plm.py [hunk 3]
```diff
@@ -0,0 +1,633 @@
...
+                )
+
+            q8_config.set_config("model.framework", "pytorch_fx")
+            model.config.save_pretrained(training_args.output_dir)
+            input_names = ["input_ids", "attention_mask","token_type_ids"]
+            model = symbolic_trace(
+                model,
+                input_names=input_names,
+                batch_size=training_args.per_device_eval_batch_size,
+                sequence_length=data_args.max_seq_length,
```
Comment (echarlaix) [inline examples/pytorch/language-modeling/run_plm.py:582]: It should be `sequence_length=max_seq_length`

File: examples/pytorch/language-modeling/run_plm.py [hunk 4]
```diff
@@ -0,0 +1,633 @@
...
+        logger.info(f"Quantized model with {metric_name} of {metric_q_model} saved to: {training_args.output_dir}")
+
+        if model_args.verify_loading:
+            from optimum.intel.neural_compressor.quantization import IncQuantizedModelForXLNetLM
+            # Load the model obtained after Intel Neural Compressor (INC) quantization
+            loaded_model = IncQuantizedModelForXLNetLM.from_pretrained(
+                training_args.output_dir,
+                input_names=input_names,
+                batch_size=training_args.per_device_eval_batch_size,
+                sequence_length=data_args.max_seq_length,
```
Comment (echarlaix) [inline examples/pytorch/language-modeling/run_plm.py:615]: It should be `sequence_length=max_seq_length`

File: examples/pytorch/summarization/run_summarization.py [hunk 5]
```diff
@@ -0,0 +1,706 @@
...
+    )
+    model_config = model.config
+    model_generate = model.generate
+
+    resume_from_checkpoint = training_args.resume_from_checkpoint
+    metric_name = model_args.tune_metric
+
+    def take_eval_steps(model, trainer, metric_name, save_metrics=False):
+        trainer.model = model
+        metrics = trainer.evaluate()
```
Comment (PenghuiCheng) [inline examples/pytorch/summarization/run_summarization.py:575]: we should  pass max_length and num_beams to the evaluate method from our Seq2SeqTrainer instance as in [run_translation.py](https://github.com/huggingface/transformers/blob/a503012275e8d2fa6e682d11c9bad68aa4c46cd6/examples/pytorch/summarization/run_summarization.py#L593)

File: examples/pytorch/language-modeling/README.md [hunk 6]
```diff
@@ -14,23 +14,28 @@ See the License for the specific language governing permissions and
...
+### GPT-2/GPT and causal language modeling
+
+The following example fine-tunes GPT-2 on WikiText-2 while applying [`inc`](https://github.com/intel/neural-compressor) 
+quantization aware training. We're using the raw WikiText-2 (no tokens were replaced before the tokenization). The loss here is that of causal language modeling (CLM). 
 
 ```bash
-python examples/pytorch/language-modeling/run_clm.py \
-    --model_name_or_path bert-base-cased \
+python run_clm.py \
+    --model_name_or_path gpt2 \
```
Comment (PenghuiCheng) [inline examples/pytorch/language-modeling/README.md:38]: For GPT2 model, the customized Conv1d ops are not in the PyTorch quantization map, so only one layer of linear will be quantized in PyTorch FBGEMM. So the benefits of quantization are relatively small for this model. I think it is not a good example here, do you think so?

Comment (echarlaix) [reply examples/pytorch/language-modeling/README.md:38]: I see your point, as CLM is unidirectional in nature, what do you think about replacing it by a GPT-Neo model (such as EleutherAI/gpt-neo-125M for example) ?

Comment (PenghuiCheng) [reply examples/pytorch/language-modeling/README.md]: Yes, I tested EleutherAI/gpt-neo-125M, and it is ok for me. I modified README and quantization.yml, please review the below commit:
https://github.com/huggingface/optimum/pull/30/commits/128d4573223ae11280262c21b71b7dbfccecdc9a
